src/sentry/models/grouphashmetadata.py

```diff
@@ -0,0 +1,64 @@
+from typing import Any
+
+from django.db import models
+from django.utils import timezone
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import FlexibleForeignKey, JSONField, Model, region_silo_model
+
+
+@region_silo_model
+class GroupHashMetadata(Model):
+    __relocation_scope__ = RelocationScope.Excluded
+
+    class GroupingMethod(models.TextChoices):
+        HASH = "hash"
+        SEER = "seer"
+
+    class HashBasis(models.TextChoices):
+        MESSAGE = "msg"
+        FULL_STACKTRACE = "stack"
+        APP_STACKTRACE = "app_stack"
+        FINGERPRINT = "fingerprint"
+        # TODO: Are there others?
+
+    # GENERAL
+    grouphash = FlexibleForeignKey("sentry.GroupHash", unique=True, related_name="metadata")
+    date_added = models.DateTimeField(default=timezone.now)
+    grouping_method = models.PositiveIntegerField(choices=GroupingMethod)
+
+    # HASHING
+    # Most recent config to produce this hash
+    # TODO: do we need first, also?
+    # TODO: Is there a way to get choices for this from grouping.strategies.configurations.CONFIGURATIONS?
+    grouping_config = models.CharField()
+    # Eg. message, in-app stacktrace, full stacktrace, fingerprint, etc
+    hash_basis = models.CharField(choices=HashBasis, null=True)
+    # Eg. if message parameterized, what type of parameterization?, etc)
+    hashing_metadata: models.Field[dict[str, Any] | None, dict[str, Any] | None] = JSONField(
+        null=True
+    )
+    # Built-in and custom stacktrace rules used
+    enhancements: models.Field[dict[str, Any] | None, dict[str, Any] | None] = JSONField(null=True)
+    # Client fingerprint, value, matched rule, rule source
+    fingerprint: models.Field[dict[str, Any] | None, dict[str, Any] | None] = JSONField(null=True)
+    # If this hash was associated to its group via a grouping config transition, what secondary hash
+    # made the link?
+    secondary_hash_match = FlexibleForeignKey(
+        "sentry.GroupHash", null=True, related_name="secondary_matches"
+    )
+
+    # SEER
+    seer_model = models.CharField(null=True)
+    # This will be different than `date_added` if we send it to Seer as part of a backfill
+    seer_date_sent = models.DateTimeField(null=True)
+    # TODO Figure out how to best reuse the types in seer.similarity.types
+    seer_results: models.Field[list[Any] | None, list[Any] | None] = JSONField(
```

Comment:
What can the seer results be?

Reply:
It'll be a list of dict versions of `SeerSimilarIssueData` instances. For now there'll only ever be one item in the list, since we're only planning to store the single best match (if any), but here and elsewhere in the code it's a list just in case that changes in the future.

https://github.com/getsentry/sentry/blob/3c02b743625293ae5b9d94b3390bda961856ae76/src/sentry/seer/similarity/types.py#L51-L56